### PR TITLE
`gpeb-entry-edit-timestamp.php`: Added snippet to display edit entry timestamp.

### PR DESCRIPTION
**Pull request by saifsultanc**
## Context

⛑️ Ticket(s): https://secure.helpscout.net/conversation/2582701663/65507?folderId=7098280

## Summary

Display the entry edit timestamp:
https://www.loom.com/share/af4e456263324971b3463ff344c77830

### CODE REVIEW

File: gp-entry-blocks/gpeb-entry-edit-timestamp.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ */
+add_action( 'gform_after_update_entry', function ( $form, $entry_id ) {
+	update_entry_meta( $entry_id );
+}, 10, 2 );
+
+add_action( 'gform_post_update_entry', function ( $entry, $original_entry ) {
+	update_entry_meta( rgar( $entry, 'id' ) );
+}, 10, 2 );
+
+function update_entry_meta( $entry_id ) {
```
Comment (claygriffiths) [inline gp-entry-blocks/gpeb-entry-edit-timestamp.php:17]: This is going to be in the global namespace so it should be scoped to this snippet.

File: gp-entry-blocks/gpeb-entry-edit-timestamp.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	$timestamp      = 'timestamp_' . $entry_id;
```
Comment (claygriffiths) [inline gp-entry-blocks/gpeb-entry-edit-timestamp.php:18]: Why does this meta key have the entry ID in it?

File: gp-entry-blocks/gpeb-entry-edit-timestamp.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	$timestamp_meta = gform_get_meta( $entry_id, $timestamp );
+	$current_user   = get_userdata( get_current_user_id() );
+	$username       = $current_user->user_login;
+	$current_value  = 'Entry updated at ' . current_datetime()->format( 'Y-m-d H:i:s' ) . ' by ' . $username . '.<br>';
+	$timestamp_meta = $current_value . $timestamp_meta;
+	gform_update_meta( $entry_id, $timestamp, $timestamp_meta );
```
Comment (claygriffiths) [inline gp-entry-blocks/gpeb-entry-edit-timestamp.php:24]:
```suggestion
     // Honestly, you could probably just utilize the `date_updated` entry property (not meta)
	gform_update_meta( $entry_id, 'last_edit_by', get_current_user_id() );
```

File: gp-entry-blocks/gpeb-entry-edit-timestamp.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+add_filter( 'gform_replace_merge_tags', function ( $text, $form, $entry, $url_encode, $esc_html, $nl2br, $format ) {
+	$custom_merge_tag = '{last_edited_by}';
+
+	if ( strpos( $text, $custom_merge_tag ) === false ) {
+		return $text;
+	}
+ 
+	$download_link = gform_get_meta( $entry['id'], 'gf_entry_editor' );
+
+	return str_replace( $custom_merge_tag, $download_link, $text );
```
Comment (claygriffiths) [inline gp-entry-blocks/gpeb-entry-edit-timestamp.php:37]: Haha, looks like you didn't replace a copy paste?

I'd do...

```
$current_user  = wp_get_current_user();

return str_replace( $custom_merge_tag, $current_user->display_name, $text );
```

File: gp-entry-blocks/gpeb-entry-edit-timestamp.php [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	$custom_merge_tag = '{entry_edit_timestamp}';
+
+	if ( strpos( $text, $custom_merge_tag ) === false ) {
+		return $text;
+	}
+	$timestamp = 'timestamp_' . $entry['id'];
+	$download_link = gform_get_meta( $entry['id'], $timestamp );
+	
+	return str_replace( $custom_merge_tag, $download_link, $text );
+}, 10, 7 );
```
Comment (claygriffiths) [inline gp-entry-blocks/gpeb-entry-edit-timestamp.php:50]: I'd just use `{date_updated}` from https://docs.gravityforms.com/entry-date-merge-tags/

**Comment by saifsultanc** [comment]
@claygriffiths Updated to Class-based snippet...

I will summarize my points here. Please let me know if I have misinterpreted anything:

1. I didn't use the `$entry['date_updated']` because that uses [UTC](https://community.gravityforms.com/t/unreliable-entry-dates-resolved/15917) and I thought it was more straightforward to directly fetch the `current_datetime` rather than getting the value from $entry data and then processing a `format_date` method call.
2. I used "entry ID" in timestamp store, because we will be having a different timestamp log for each entry. Anything I am missing here? Please let me know.
3. Sorry about those silly variable names, I was copying from Gravity Forms docs (template) and didnt realize I missed updating those names 😬 
4. PHPCS all clear, sorry about that.

**Comment by claygriffiths** [comment]
I definitely saw what David said about making something generic, but I was looking back at the original request, and it looks like they originally wanted to leverage GravityRevisions. No biggie though, I think this will be a valuable snippet, regardless.

Regarding the edit time, storing UTC in the database is ideal because it’s easy to convert it to timezones. I’d completely ditch anything concerning the timestamp in the snippet. The WP entry date merge tags will be converted into the timezone and formats configured in WP by default: https://docs.gravityforms.com/entry-date-merge-tags/#:~:text=The%20values%20will%20be%20formatted%20for%20output%20using%20the%20timezone%2C%20date%2C%20and/or%20time%20formats%20from%20the%20WordPress%20General%20Settings%20screen.

As far as 'timestamp_' . $entry_id as the meta key goes, the meta is already specific to the entry so I’m struggling to figure out why appending the entry ID is needed over something like gw_last_edit_timestamp. That aside, I’d still ditch it in favor of the built in {date_updated} unless I’m missing something.

**Comment by claygriffiths** [comment]
Sorry, I now see that you're trying to keep a log of edits.

**Review by claygriffiths — APPROVED** [review]
LGTM!

**Pull request merged**